src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -329,14 +329,27 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
   // same TaskID to request a worker
   auto resource_spec_msg = scheduling_key_entry.resource_spec.GetMutableMessage();
   resource_spec_msg.set_task_id(TaskID::FromRandom(job_id_).Binary());
-  const TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
+  TaskSpecification resource_spec = TaskSpecification(resource_spec_msg);
   rpc::Address best_node_address;
   const bool is_spillback = (raylet_address != nullptr);
+  float scheduler_spread_threshold = RayConfig::instance().scheduler_spread_threshold();
   if (raylet_address == nullptr) {
+    bool is_selected_based_on_locality;
     // If no raylet address is given, find the best worker for our next lease request.
-    best_node_address = lease_policy_->GetBestNodeForTask(resource_spec);
+    std::tie(best_node_address, is_selected_based_on_locality) =
+        lease_policy_->GetBestNodeForTask(resource_spec);
+    if (is_selected_based_on_locality) {
+      scheduler_spread_threshold = 1.0;
+    }
     raylet_address = &best_node_address;
   }
+  if (resource_spec.GetMessage().scheduling_strategy().scheduling_strategy_case() ==
+      rpc::SchedulingStrategy::SchedulingStrategyCase::kDefaultSchedulingStrategy) {
```

Comment:
What about the placement group case?

Reply:
update: I changed the PR so placement group case also runs the locality logic as before.